ddtrace/internal/_encoding.pyx

```diff
@@ -288,8 +296,11 @@ cdef class Packer(object):
         ret = msgpack_pack_array(&self.pk, L)
         if ret != 0: raise RuntimeError("Couldn't pack trace")
 
+        if L > 0 and trace[0].context is not None:
+            dd_origin = trace[0].context.dd_origin
+
         for span in trace:
-            ret = self._pack_span(span)
+            ret = self._pack_span(span, dd_origin)
```

Comment:
`dd_origin` is possibly undefined here?

Reply:
```suggestion
            if dd_origin is not None:
                span.meta[ORIGIN_KEY] = dd_origin
            ret = self._pack_span(span)
```

This is what I had in mind instead of changing `_pack_span` or `_pack_meta`.

Reply:
this seems reasonable to me to try.

I still like the idea of handling dd_origin in the encoder, or at least on the writer side, since how we send to the backend could change.

doing it in the encoder will be helpful for v0.5 trace agent endpoint since we can optimize how we put it into the string table up front.